lasagne/layers/autoenc.py

```diff
@@ -0,0 +1,262 @@
+import theano
+
+from .. import nonlinearities
+from .. import init
+
+from .base import MergeLayer
+from .helper import get_all_layers
+from .input import InputLayer
+from .special import NonlinearityLayer, BiasLayer
+from .noise import DropoutLayer, GaussianNoiseLayer
+
+
+__all__ = [
+    "InverseLayer",
+    "build_autoencoder",
+    "unfold_bias_and_nonlinearity_layers",
+]
+
+
+class InverseLayer(MergeLayer):
+    """
+    The :class:`InverseLayer` class performs inverse operations
+    for a single layer of a neural network by applying the
+    partial derivative of the layer to be inverted with
+    respect to its input: transposed layer
+    for a :class:`DenseLayer`, deconvolutional layer for
+    :class:`Conv2DLayer`, :class:`Conv1DLayer`; or
+    an unpooling layer for :class:`MaxPool2DLayer`.
+
+    It is specially useful for building (convolutional)
+    autoencoders with tied parameters.
+
+    Note that if the layer to be inverted contains a nonlinearity
+    and/or a bias, the :class:`InverseLayer` will include the derivative
+    of that in its computation.
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape.
+    layer : a :class:`Layer` instance or a tuple
```

Comment:
"or a tuple" is wrong here, actually. It has to be a Layer instance, otherwise it won't work.